src/Evaluators/common.jl

```diff
@@ -50,6 +85,17 @@ struct AutoDiffFactory{VT} <: AbstractAutoDiffFactory
     ∇f::AdjointStackObjective{VT}
 end
 
+struct HessianFactory{SpMT}
+    hashu::UInt64
```

Comment:
What is this?

Reply:
It just stores the hash for the last variable `u` evaluated. By doing that, we avoid computing several times the Hessian for a same variable `u` (when evaluating hessian vector product). That should be removed once the new AD backend for Hessian is implemented.

Reply:
Got it.